go/oasis-node/cmd/node/node.go

```diff
@@ -166,6 +167,50 @@ func (n *Node) RequestShutdown() (<-chan struct{}, error) {
 	return n.RegistrationWorker.Quit(), nil
 }
 
+func (n *Node) Ready() <-chan struct{} {
+	return n.readyCh
+}
+
+func (n *Node) WaitReady() error {
```

Comment:
Why is this method public, it seems to be used internally and calling it from the outside can cause a panic due to closing an already closed channel? Also my suggestion is to change this to not return an error, log it instead and below just call `go n.waitReady()` instead of having that wrapper closure.